extensions/amp-a4a/0.1/test/test-streaming-utils.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {macroTask} from '../../../../testing/yield';
+import {streamResponseToWriter} from '../streaming-utils';
+
+const chunk1 = `
+ <!doctype html>
+ <html ⚡4ads>
+ <head>
+   <meta charset="utf-8">
+   <meta name="viewport" content="width=device-width,minimum-scale=1">
+`;
+
+const chunk2 = `
+   <style amp4ads-boilerplate>body{visibility:hidden}</style>
+   <script async src="https://cdn.ampproject.org/amp4ads-v0.js"></script>
+ </head>
+ <body>Hello, AMP4ADS world.</body>
+ </html>
+`;
+
+describes.fakeWin('streamResponseToWriter', {}, (env) => {
```

Comment:
How do we want to handle an interrupted stream?

Reply:
Do you mean if the browser cancels the response? I'm not sure, but if that is the same as calling `.cancel` on the stream, the writer will be closed and if there is no content we will collapse the ad. If there is a body I think we will render a partial ad. If there is no `cancel` called I think we will just wait until we receive a body, though maybe we should time out. Not sure I am answering your question, let me know if you want to chat about this.